src/rviz/default_plugin/marker_utils.cpp

```diff
@@ -76,4 +84,346 @@ MarkerBase* createMarker(int marker_type, MarkerDisplay* owner, DisplayContext*
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::SPHERE_LIST:
+  case visualization_msgs::Marker::CUBE_LIST:
+  case visualization_msgs::Marker::TRIANGLE_LIST:
+    checkQuaternion(marker, ss);
+    checkScale(marker, ss);
+    checkPointsNotEmpty(marker, ss);
+    checkColors(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::POINTS:
+    checkScalePoints(marker, ss);
+    checkPointsNotEmpty(marker, ss);
+    checkColors(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::TEXT_VIEW_FACING:
+    checkColor(marker, ss);
+    checkScaleText(marker, ss);
+    checkTextNotEmptyOrWhitespace(marker, ss);
+    checkPointsEmpty(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::MESH_RESOURCE:
+    checkQuaternion(marker, ss);
+    checkColor(marker, ss);
+    checkScale(marker, ss);
+    checkPointsEmpty(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkTextEmpty(marker, ss);
+    break;
+
+  default:
+    ss << "Unknown marker type: " << marker.type ;
+  }
+
+  if(ss.tellp() != 0) //stringstream is not empty
+  {
+    std::string warning = ss.str();
+
+    owner->setMarkerStatus(MarkerID(marker.ns, marker.id), StatusProperty::Warn, warning);
+    ROS_WARN("Marker '%s/%d': %s", marker.ns.c_str(), marker.id, warning.c_str());
+    return false;
+  }
+
+  owner->deleteMarkerStatus(MarkerID(marker.ns, marker.id));
+  return true;
+}
+
+bool checkMarkerArrayMsg(const visualization_msgs::MarkerArray& array, MarkerDisplay* owner)
+{
+  std::vector<MarkerID> marker_ids;
+
+  bool add_marker_in_array = false;
+
+  for(int i = 0; i<array.markers.size(); i++)
+  {
+    if(array.markers[i].action == visualization_msgs::Marker::ADD)
+    {
+      add_marker_in_array = true;
+    }
+
+    if(add_marker_in_array && i != 0 && array.markers[i].action == visualization_msgs::Marker::DELETEALL)
+    {
+      std::stringstream warning;
+      warning << "found a DELETEALL at index " << i << ", previous markers in the MarkerArray will never show";
+      ROS_WARN("MarkerArray: %s", warning.str().c_str());
+      owner->setStatusStd(StatusProperty::Warn, "marker_array", warning.str());
+      return false;
+    }
+    MarkerID current_id(array.markers[i].ns, array.markers[i].id);
+    std::vector<MarkerID>::iterator search = std::lower_bound(marker_ids.begin(), marker_ids.end(), current_id);
+    if (search != marker_ids.end())
+    {
+      std::stringstream ss;
+      ss << "found '" <<  array.markers[i].ns.c_str() << "/" << array.markers[i].id << "' multiple times";
+      ROS_WARN("MarkerArray: %s", ss.str().c_str());
+      owner->setStatusStd(StatusProperty::Warn, "marker_array", ss.str());
+      return false;
+    }
+    else
+    {
+      marker_ids.insert(search, current_id);
+    }
+  }
```

Comment:
Shouldn't there be a `checkMarkerMsg(array.markers[i])` here as well to check the individual markers?

Reply:
Individual markers in a MarkerArray are already checked because they are added to the tf::MessageFilter which calls the incomingMarker function. The callback is registered [here](https://github.com/TAMS-Group/rviz/blob/c028953511c959d88d266fc0f971e6decc1bc09b/src/rviz/default_plugin/marker_display.cpp#L81)